samples/client/petstore/swift5/vaporLibrary/Sources/PetstoreClient/APIs/StoreAPI.swift

```diff
@@ -0,0 +1,228 @@
+            try Configuration.apiWrapper(&request)
+            
+            
+            
+            try beforeSend(&request)
+        }
+    }
+
+    public enum GetOrderById {
+        case http200(value: Order, raw: ClientResponse)
+        case http400(value: Void, raw: ClientResponse)
+        case http404(value: Void, raw: ClientResponse)
+        case http0(value: Order, raw: ClientResponse)
+    }
+
+    /**
+     Find purchase order by ID
+     GET /store/order/{order_id}
+     For valid response try integer IDs with value <= 5 or > 10. Other values will generated exceptions
+     - parameter orderId: (path) ID of pet that needs to be fetched 
+     - returns: `EventLoopFuture` of `GetOrderById` 
+     */
+    open class func getOrderById(orderId: Int64, headers: HTTPHeaders = PetstoreClient.customHeaders, beforeSend: (inout ClientRequest) throws -> () = { _ in }) -> EventLoopFuture<GetOrderById> {
+        return getOrderByIdRaw(orderId: orderId, headers: headers, beforeSend: beforeSend).flatMapThrowing { response -> GetOrderById in
+            switch response.status.code {
+            case 200:
+                return .http200(value: try response.content.decode(Order.self, using: Configuration.contentConfiguration.requireDecoder(for: Order.defaultContentType)), raw: response)
+            case 400:
+                return .http400(value: (), raw: response)
+            case 404:
+                return .http404(value: (), raw: response)
+            default:
+                return .http0(value: try response.content.decode(Order.self, using: Configuration.contentConfiguration.requireDecoder(for: Order.defaultContentType)), raw: response)
+            }
+        }
+    }
+
+
+    /**
+     Place an order for a pet
+     POST /store/order
+     - parameter body: (body) order placed for purchasing the pet 
+     - returns: `EventLoopFuture` of `ClientResponse` 
+     */
+    open class func placeOrderRaw(body: Order, headers: HTTPHeaders = PetstoreClient.customHeaders, beforeSend: (inout ClientRequest) throws -> () = { _ in }) -> EventLoopFuture<ClientResponse> {
+        let path = "/store/order"
+        let URLString = PetstoreClient.basePath + path
+
+        guard let apiClient = Configuration.apiClient else {
+            fatalError("Configuration.apiClient is not set.")
+        }
+
+        return apiClient.send(.POST, headers: headers, to: URI(string: URLString)) { request in
+            try Configuration.apiWrapper(&request)
+            
+            
+            try request.content.encode(body, using: Configuration.contentConfiguration.requireEncoder(for: Order.defaultContentType))
+            
+            try beforeSend(&request)
+        }
+    }
+
+    public enum PlaceOrder {
+        case http200(value: Order, raw: ClientResponse)
+        case http400(value: Void, raw: ClientResponse)
+        case http0(value: Order, raw: ClientResponse)
```

Comment:
@aymanbagabas I think at least the default case, should have an int with the status code, because it can be something that may be important for debug proposes like 401 or 403 for example.
And Maybe instead of `http0` it could be `httpUnknown` to be more clearer?

Reply:
[`ClientResponse`](https://api.vapor.codes/vapor/main/Vapor/ClientResponse/#clientresponse.status) type has a `.status` property that has the response code. I agree, `httpUnknown` is more clearer. Also, I think that case should optionally decode the response since we don't really know what the response would look like.

Reply:
You are right, I haven't noticed that ClientResponse has http status code, headers, etc, that's nice 👍

Reply:
yeah, Vapor is a really cool framework, it forces you to write beautiful code. You should try it sometime. There's [vapor-server-codegen](https://github.com/thecheatah/vapor-server-codegen), I think bringing that to the OpenAPI generator would be a great idea.

Reply:
When you have time, please try to change `http0` to `httpUnknown` and check the `value` Type in each enum case, to see what makes more sense, for us to merge this 🙂